include/dicom/dicom.h

```diff
@@ -220,1618 +272,586 @@ typedef struct _DcmError DcmError;
  */
 DCM_EXTERN
-const char *dcm_element_get_value_CS(const DcmElement *element,
-                                     uint32_t index);
+uint32_t dcm_dict_tag_from_keyword(const char *keyword);
 
 /**
- * Get value of a Data Element with Value Representation DA (Date).
+ * Find the VR for a tag. This will return DCM_VR_ERROR if the tag is unknown,
+ * or does not have a unique VR.
  *
- * :param element: Pointer to Data Element
- * :param index: Zero-based index of value within the Data Element
+ * :param tag: Attribute Tag
  *
- * :return: Pointer to memory location where value is stored
+ * :return: the unique VR for this tag, or DCM_VR_ERROR
  */
 DCM_EXTERN
-const char *dcm_element_get_value_DA(const DcmElement *element,
-                                     uint32_t index);
+DcmVR dcm_vr_from_tag(uint32_t tag);
 
 /**
- * Get value of a Data Element with Value Representation DS (Decimal String).
+ * Determine whether a Tag is public.
  *
- * :param element: Pointer to Data Element
- * :param index: Zero-based index of value within the Data Element
+ * A Tag is public if it is defined in the Dictionary.
+ *
+ * :param tag: Attribute Tag
  *
- * :return: Pointer to memory location where value is stored
+ * :return: Yes/no answer
  */
 DCM_EXTERN
-const char *dcm_element_get_value_DS(const DcmElement *element,
-                                     uint32_t index);
+bool dcm_is_public_tag(uint32_t tag);
 
 /**
- * Get value of a Data Element with Value Representation DT (Date Time).
+ * Determine whether a Tag is private.
  *
- * :param element: Pointer to Data Element
- * :param index: Zero-based index of value within the Data Element
+ * :param tag: Attribute Tag
  *
- * :return: Pointer to memory location where value is stored
+ * :return: Yes/no answer
  */
 DCM_EXTERN
-const char *dcm_element_get_value_DT(const DcmElement *element,
-                                     uint32_t index);
+bool dcm_is_private_tag(uint32_t tag);
 
 /**
- * Get value of a Data Element with Value Representation FD
- * (Floating Point Double).
+ * Determine whether a Tag is valid.
  *
- * :param element: Pointer to Data Element
- * :param index: Zero-based index of value within the Data Element
+ * :param tag: Attribute Tag
  *
- * :return: Value
+ * :return: Yes/no answer
  */
 DCM_EXTERN
-double dcm_element_get_value_FD(const DcmElement *element,
-                                uint32_t index);
+bool dcm_is_valid_tag(uint32_t tag);
 
 /**
- * Get value of a Data Element with Value Representation FL (Floating Point).
+ * Determine whether a Value Representation is valid.
  *
- * :param element: Pointer to Data Element
- * :param index: Zero-based index of value within the Data Element
+ * :param vr: Attribute Value Representation
  *
- * :return: Value
+ * :return: Yes/no answer
  */
 DCM_EXTERN
-float dcm_element_get_value_FL(const DcmElement *element,
-                               uint32_t index);
+bool dcm_is_valid_vr(const char *vr);
 
```

Comment:
Internally it's using an enum for this:

```C
typedef enum _DcmVRClass {
    DCM_CLASS_ERROR,
    DCM_CLASS_STRING_MULTI,
    DCM_CLASS_STRING_SINGLE,
    DCM_CLASS_NUMERIC,
    DCM_CLASS_BINARY,
    DCM_CLASS_SEQUENCE
} DcmVRClass;
```

With the class describing the general nature of the VR. I found an enum more convenient than a set of testers, since you can `switch()` on it. I was planning to expose this in the public API in the next version, since it would be very useful for the openslide connector.

Reply:
Ok, let's discuss how to best expose that functionality publicly in the next version.